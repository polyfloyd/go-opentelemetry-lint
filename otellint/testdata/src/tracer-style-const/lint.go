package lint

import (
	"context"
	"fmt"
	"net/http"
)

const tracerName = "test"

func MissingSpan(ctx context.Context) { // want "Missing OpenTelemetry span"
	fmt.Println("hi")
}

func HTTPHanderMissingSpan(w http.ResponseWriter, r *http.Request) { // want "Missing OpenTelemetry span"
	fmt.Fprintf(w, "hi")
}
