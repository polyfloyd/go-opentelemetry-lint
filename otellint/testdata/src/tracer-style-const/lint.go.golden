package lint

import (
	"context"
	"fmt"
	"net/http"
)

const tracerName = "test"

func MissingSpan(ctx context.Context) { // want "Missing OpenTelemetry span"
	ctx, span := otel.Tracer(tracerName).Start(ctx, "MissingSpan")
	defer span.End()

	fmt.Println("hi")
}

func HTTPHanderMissingSpan(w http.ResponseWriter, r *http.Request) { // want "Missing OpenTelemetry span"
	ctx, span := otel.Tracer(tracerName).Start(r.Context(), "HTTPHanderMissingSpan")
	defer span.End()

	fmt.Fprintf(w, "hi")
}
